lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -4,42 +4,102 @@ module Floe
+
+          token   = options["token"]
+          token ||= File.read(options["token_file"]) if options.key?("token_file")
+          token ||= File.read(TOKEN_FILE) if File.exist?(TOKEN_FILE)
+
+          server   = options["server"]
+          server ||= URI::HTTPS.build(:host => ENV.fetch("KUBERNETES_SERVICE_HOST"), :port => ENV.fetch("KUBERNETES_SERVICE_PORT", 6443)) if ENV.key?("KUBERNETES_SERVICE_HOST")
+
+          ca_file   = options["ca_file"]
+          ca_file ||= CA_CERT_FILE if File.exist?(CA_CERT_FILE)
+
+          verify_ssl = options["verify_ssl"] == "false" ? OpenSSL::SSL::VERIFY_NONE : OpenSSL::SSL::VERIFY_PEER
+
+          if server && token
+            ssl_options = {:verify_ssl => verify_ssl}
+            ssl_options[:ca_file] = ca_file if ca_file
+
+            auth_options = {:bearer_token => token}
+          elsif kubecontext
+            server       = kubecontext.api_endpoint
+            ssl_options  = kubecontext.ssl_options
+            auth_options = kubecontext.auth_options
+          else
+            raise ArgumentError, "Missing connections options, provide a kubeconfig file or pass server and token via --docker-runner-options"
+          end
+
+          @namespace  = options.fetch("namespace", "default")
+
+          @kubeclient = Kubeclient::Client.new(server, "v1", :ssl_options => ssl_options, :auth_options => auth_options)
+          @kubeclient.discover
 
           super
         end
 
         def run!(resource, env = {}, secrets = {})
           raise ArgumentError, "Invalid resource" unless resource&.start_with?("docker://")
 
-          image     = resource.sub("docker://", "")
-          name      = pod_name(image)
-          secret    = create_secret!(secrets) if secrets && !secrets.empty?
-          overrides = pod_spec(image, env, secret)
+          image  = resource.sub("docker://", "")
+          name   = pod_name(image)
+          secret = create_secret!(secrets) if secrets && !secrets.empty?
 
-          result = kubectl_run!(image, name, overrides)
+          begin
+            create_pod!(name, image, env, secret)
+            while running?(name)
+              sleep(1)
+            end
 
-          # Kubectl prints that the pod was deleted, strip this from the output
-          output = result.output.gsub(/pod "#{name}" deleted/, "")
+            exit_status = success?(name) ? 0 : 1
+            results     = output(name)
 
-          [result.exit_status, output]
-        ensure
-          delete_secret!(secret) if secret
+            [exit_status, results]
+          ensure
+            cleanup(name, secret)
+          end
+        end
+
+        def running?(pod_name)
+          %w[Pending Running].include?(pod_info(pod_name).dig("status", "phase"))
+        end
+
+        def success?(pod_name)
+          pod_info(pod_name).dig("status", "phase") == "Succeeded"
+        end
+
+        def output(pod)
+          kubeclient.get_pod_log(pod, namespace).body
```

Comment:
NOTE I think it is weird that `kubeclient` returns a `RestClient` object here instead of the body like the other calls.  Need to see if there was a reason for that or if it was an oversight.

Particularly because they move to Faraday in master so the next major release this would have to change.